internal/pkg/install/install.go

```diff
@@ -205,19 +216,37 @@ func RunInstallerContainer(disk, platform, ref string, cfg config.Provider, opts
 
 	w := &kmsg.Writer{KmsgWriter: f}
 
-	configR := &containerdrunner.StdinCloser{
-		Stdin:  bytes.NewReader(configBytes),
-		Closer: make(chan struct{}),
+	var (
+		r       io.Reader
+		configR *containerdrunner.StdinCloser
+	)
+
+	if cfg != nil {
+		var configBytes []byte
+
+		configBytes, err = cfg.Bytes()
+		if err != nil {
+			return err
+		}
+
+		configR = &containerdrunner.StdinCloser{
+			Stdin:  bytes.NewReader(configBytes),
+			Closer: make(chan struct{}),
+		}
+
+		r = configR
 	}
 
-	creator := cio.NewCreator(cio.WithStreams(configR, w, w))
+	creator := cio.NewCreator(cio.WithStreams(r, w, w))
```

Comment:
Can't we use `configR` here?

Reply:
`r` is `io.Reader`, so `r == nil` will be true (if it's `nil`)
`configR` is `*StdinCloser`, so if it's `nil`, then if passed as `io.Reader` it won't compare to `nil` (`r != nil`)

Reply:
Maybe then 
```go
	var r interface {
		io.Reader
		WaitAndClose(ctx context.Context, task containerd.Task)
	}
```

Reply:
this should probably work

Reply:
fixed, thanks!